src/main/java/Duke.java

```diff
@@ -1,10 +1,301 @@
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException("\t ☹ OOPS!!! The description or time of an event cannot be empty.\n");
+            }
+            if (description.equals("") || time.equals("")) {
+                throw new DukeException("\t ☹ OOPS!!! The description or time of an event cannot be empty.\n");
+            }
+            t = new Event(description, this.strToDate(time));
+        } else { 
+            // deadline
+            timeIndex = input.indexOf(" /by ");
+            if (timeIndex == -1) { 
+                // cannot find '/at'
+                throw new DukeException("\t ☹ OOPS!!! The time of a deadline cannot be empty.\n");
+            }
+            try {
+                description = input.substring(9, timeIndex);
+                time = input.substring(timeIndex + 5);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException("\t ☹ OOPS!!! The description or time of a deadline cannot be empty.\n");
+            }
+            if (description.equals("") || time.equals("")) {
+                throw new DukeException("\t ☹ OOPS!!! The description or time of a deadline cannot be empty.\n");
+            }
+            t = new Deadline(description, this.strToDate(time));
+        }
+        this.lst.add(t);
+        this.printMsg("\t Got it. I've added this task:\n\t   " + t.toString()
+                + "\n\t Now you have " + this.lst.size() + " tasks in the list.\n");
+    }
+
+    /**
+     * Marks a particular task as done.
+     *
+     * @param num The number of the task to be marked.
+     */
+    private void mark(int num) {
+        this.lst.get(num).markAsDone();
+        this.printMsg("\t Nice! I've marked this task as done:\n\t   " + lst.get(num).toString() + "\n"); 
+    }
+
+    /**
+     * Unmarks a particular task as done.
+     *
+     * @param num The number of the task to be unmarked.
+     */
+    private void unmark(int num) {
+        this.lst.get(num).unmarkAsDone();
+        this.printMsg("\t OK, I've marked this task as not done yet:\n\t   " + lst.get(num).toString() + "\n");
+    }
+
+    /**
+     * Deletes a task in the specified index.
+     *
+     * @param num The number of the task to be removed.
+     */
+    private void delete(int num) {
+        Task t = this.lst.remove(num);
+        this.printMsg("\t Noted. I've removed this task:\n\t   " + t.toString()
+                + "\n\t Now you have " + this.lst.size() + " tasks in the list.\n");
+    }
+
+    /**
+     * Gives the bot intructions and let it do the corresponding job.
+     */
+    private void run() {
+        String input = sc.nextLine();
+        String[] splitted = input.split("\\s+");
+        if (input.equals("bye")) { 
+            return;
+        } else if (input.equals("list")) {
+            this.list();
+        } else if (splitted.length == 2 && splitted[0].equals("mark")) {
```

Comment:
Perhaps can consider making splitted[0] a variable since it is used quite often in this code section.